packages/@react-spectrum/dialog/src/AlertDialog.tsx

```diff
@@ -64,49 +65,60 @@ function AlertDialog(props: SpectrumAlertDialogProps, ref: DOMRef) {
     }
   }
 
+  let onKeyDown = (e) => {
+    if (e.key === 'Enter' && allowsKeyboardConfirmation) {
+      e.stopPropagation();
+      e.preventDefault();
+      onPrimaryAction();
+      onClose();
+    }
+  };
+
   return (
-    <Dialog
-      UNSAFE_style={styleProps.style}
-      UNSAFE_className={classNames(styles, {[`spectrum-Dialog--${variant}`]: variant}, styleProps.className)}
-      isHidden={styleProps.hidden}
-      size="M"
-      role="alertdialog"
-      ref={ref}>
-      <Heading>{title}</Heading>
-      {(variant === 'error' || variant === 'warning') &&
-        <AlertMedium
-          slot="typeIcon"
-          aria-label={formatMessage('alert')} />
-      }
-      <Divider />
-      <Content>{children}</Content>
-      <ButtonGroup align="end">
-        {cancelLabel &&
-          <Button
-            variant="secondary"
-            onPress={() => chain(onClose(), onCancel())}
-            autoFocus={autoFocusButton === 'cancel'}>
-            {cancelLabel}
-          </Button>
+    <AlertDialogContext.Provider value={{onKeyDown}}>
```

Comment:
I use a provider to pass the event handler down to `Dialog` so that we don't need to change the `Dialog` interface definition. I'm not sure if there's a better approach, any suggestions are welcome!